lib/puppet/resource/type.rb

```diff
@@ -413,6 +413,9 @@ def assign_defaults(resource, scope, resource_hash)
 
   def validate_resource_hash(resource, resource_hash)
     Puppet::Pops::Types::TypeMismatchDescriber.validate_parameters(resource.to_s, parameter_struct, resource_hash)
+  rescue Puppet::ParseError => e
```

Comment:
It's possible to test half of it. There's no backtrace unless an error is rescued. Also testing something that is an exceptional case (environment compiler + consume meta parameter) everytime some resource parameters are validated is seems rather unnecessary.